src/arraytypes/struct.jl

```diff
@@ -33,23 +33,33 @@ isnamedtuple(T) = false
 istuple(::Type{<:Tuple}) = true
 istuple(T) = false
 
-@propagate_inbounds function Base.getindex(s::Struct{T,S}, i::Integer) where {T,S}
+if isdefined(ArrowTypes, :StructElement)
```

Comment:
Went with this approach to avoid needing to change Arrow's compat bounds on ArrowTypes